File: src/main/java/com/example/demo/user/profile/dto/UserProfileResponse.java
package com.example.demo.user.profile.dto;

import java.time.Instant;

public record UserProfileResponse(
        String id,
        String username,
        String nickname,
        Instant createdAt,
        Instant updatedAt
) {
    private static final String EMPTY_NICKNAME = "";

    public UserProfileResponse {
        if (nickname == null) {
            nickname = EMPTY_NICKNAME;
        }
    }
}

File: src/main/java/com/example/demo/user/profile/dto/UserProfileResponse.kt
package com.example.demo.user.profile.dto

import java.time.Instant

class UserProfileResponse(
    val id: String,
    val username: String,
    nickname: String?,
    val createdAt: Instant,
    val updatedAt: Instant,
) {
    val nickname: String = nickname ?: EMPTY_NICKNAME

    companion object {
        private const val EMPTY_NICKNAME = ""
    }
}
